### PR TITLE
Cleanup 183

### PR DESCRIPTION
**Pull request by samuelcolvin**
a few things from #183 which were easier to implement directly than explain.

Also fix coverage.

### CODE REVIEW

**Comment by samuelcolvin** [comment]
@Gr1N, please have a look and confirm you're happy with this since it's mostly changes to your code.

If so I'll merge.

File: pydantic/utils.py [hunk 1]
```diff
@@ -125,7 +127,12 @@ def display_as_type(v):
...
 
 def to_snake_case(v: str) -> str:
-    v = re.sub(r'([A-Z]+)([A-Z][a-z])', r'\1_\2', v)
-    v = re.sub(r'([a-z\d])([A-Z])', r'\1_\2', v)
+    return SNAKE_CASE_REGEX.sub(r'\1\3_\2\4', v).lower()
 
-    return v.replace('-', '_').lower()
+
+@contextmanager
+def change_exception(raise_exc, *except_types):
```
Comment (Gr1N) [inline pydantic/utils.py]: Great improvement! But I think better to use another signature for func  `change_exception(raise_exc, class_or_tuple)` like in `isinstance` or `issubclass`, with this more easily to understand what you what to catch and what to raise when use this func.

Comment (samuelcolvin) [reply pydantic/utils.py]: done.

Comment (Gr1N) [reply pydantic/utils.py]: You changed only name, not signature

Comment (samuelcolvin) [reply pydantic/utils.py]: Oh, I see what you mean, I've always really disliked the signature of `isinstance` adn `issubclass`, I'd rather keep it like this.

Comment (Gr1N) [reply pydantic/utils.py]: Ok :) but better to rename back arg name then

Comment (samuelcolvin) [reply pydantic/utils.py]: yes, I was being dumb.

**Comment by Gr1N** [comment]
No more questions, let’s merge!

**Pull request merged**